src/SubscribedStream.js

```diff
@@ -6,30 +6,60 @@ export default class SubscribedStream {
         this._client = client
         this.streamId = streamId
         this.subscriptions = {}
+        this.isPublisherPromises = {}
     }
 
-    getPublishers() {
+    async getPublishers() {
         if (!this.publishersPromise || (Date.now() - this.lastAccess) > PUBLISHERS_EXPIRATION_TIME) {
```

Comment:
If a cache gets hit on a regular basis (that is `<= PUBLISHERS_EXPIRATION_TIME`), it means that the `this._client.getStreamPublishers`will not be invoked? Would it be better to have time-based expiration so that we have a guarantee that it gets updated every now and then regardless of access patterns?

Reply:
The cache gets updated every `PUBLISHERS_EXPIRATION_TIME` milliseconds in any case because `this.lastAccess` is updated only inside this `if` statement, not for every cache hit. But maybe this variable should be renamed to `this.lastUpdated` to avoid confusion.

Reply:
Ah indeed true. Rename would probably be good idea.